Makefile.toml

```diff
@@ -266,21 +266,33 @@ cargo install \
 '''
 ]
 
-[tasks.build-packages]
+# Builds a package including its build-time and runtime dependency packages.
+[tasks.build-package]
 dependencies = ["build-tools", "publish-setup"]
+script_runner = "bash"
 script = [
 '''
+set -e
+if [ -z "${PACKAGE}" ]; then
+    echo "the PACKAGE environment variable must be set. for example:"
+    echo "cargo make build-package -e PACKAGE=kernel"
+    exit 1
+fi
 export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
 cargo build \
   ${CARGO_BUILD_ARGS} \
   ${CARGO_MAKE_CARGO_ARGS} \
-  --manifest-path packages/Cargo.toml \
-  --all
+  --manifest-path "variants/Cargo.toml" \
+  --package "${PACKAGE}"
```

Comment:
Does this work? In my scratch workspace, it seems like only packages that are part of the workspace are valid targets.

```
fedora on ip-10-0-60-129 in ~/scratch/ugly 
❯ cargo build --manifest-path variants/Cargo.toml --package foo
error: package ID specification `foo` matched no packages

fedora on ip-10-0-60-129 in ~/scratch/ugly 
❯ cargo build --manifest-path variants/Cargo.toml --package bar
error: package ID specification `bar` matched no packages

fedora on ip-10-0-60-129 in ~/scratch/ugly 
❯ cargo build --manifest-path variants/Cargo.toml --package baz
   Compiling foo v0.1.0 (/home/fedora/scratch/ugly/packages/foo)
   Compiling bar v0.1.0 (/home/fedora/scratch/ugly/packages/bar)
   Compiling baz v0.1.0 (/home/fedora/scratch/ugly/variants/baz)
    Finished dev [unoptimized + debuginfo] target(s) in 0.76s
```

Reply:
I think it did work when I ran cargo make build-package -ePACKAGE=systemd. Though I can’t explain why that would work or why I expected it to. I’ll take a closer look. Maybe it works so long as —package is a transient dependency of a workspace? 🤔

Reply:
This ended up being a difference in behavior between `cargo` from 1.49.0 and 1.51.0 - the newer version works fine.